port/zephyr/golioth_fw_zephyr.c

```diff
@@ -221,25 +242,90 @@ enum golioth_status fw_update_handle_block(const uint8_t *block,
     return GOLIOTH_OK;
 }
 
-void fw_update_post_download(void)
+enum golioth_status fw_update_post_download(void)
 {
     int err;
 
     if (!_flash_img_context.stream.buf_bytes)
     {
-        return;
+        return GOLIOTH_OK;
     }
 
     err = flash_img_buffered_write(&_flash_img_context, NULL, 0, true);
     if (err)
     {
         LOG_ERR("Failed to write to flash: %d", err);
+        return GOLIOTH_ERR_IO;
     }
+
+    return GOLIOTH_OK;
 }
 
-enum golioth_status fw_update_validate(void)
+/*
+ * @note This is similar to Zephyr's flash_img_check() but uses the Golioth
+ * port's sha256 API.
+ */
```

Comment:
In addition to being consistent with our sha256 implementation, this is also necessary because building with `flash_img_check` enabled with mbedtls is currently broken on nordic platforms due to https://github.com/zephyrproject-rtos/zephyr/pull/85238